src/Microsoft.ML.Core/Data/SchemaDebuggerProxy.cs

```diff
@@ -19,7 +19,7 @@ internal sealed class SchemaDebuggerProxy
 
         public SchemaDebuggerProxy(Schema schema)
         {
-            Columns = Enumerable.Range(0, schema.ColumnCount).Select(x => schema[x]).ToArray();
+            Columns = Enumerable.Range(0, schema.Count).Select(x => schema[x]).ToArray();
```

Comment:
>Enumerable.Range(0, schema.Count).Select(x => schema[x]).ToArray(); [](start = 22, length = 67)

I feel like we can wait to do this later, but let's keep in mind now that the schema actually is a read-only-list that this code here can be improved.

Reply:
True. I started to faithfully convert usages of `ColumnCount`, but got overwhelmed by the quantity, so I gave up for now. Next PRs will be covering that, where I'm moving from the old interface of `Schema` to the new one, and fixing usages.

---
In reply to: [235106067](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235106067) [](ancestors = 235106067)

Reply:
Not even column count, here you could just do `schema.ToArray()` I think? :)

---
In reply to: [235128149](https://github.com/dotnet/machinelearning/pull/1680#discussion_r235128149) [](ancestors = 235128149,235106067)